src/data/nat/prime.lean

```diff
@@ -1150,6 +1164,15 @@ end
 lemma mem_factors_mul_right {p a b : ℕ} (hpb : p ∈ b.factors) (ha : a ≠ 0) : p ∈ (a*b).factors :=
 by { rw mul_comm, exact mem_factors_mul_left hpb ha }
 
+lemma exists_odd_prime_and_dvd_or_two_pow (n : ℕ) :
+  (∃ k : ℕ, n = 2 ^ k) ∨ ∃ p, nat.prime p ∧ p ∣ n ∧ odd p :=
```

Comment:
The name and the statement don't agree on the order of assertions
```suggestion
lemma exists_odd_prime_and_dvd_or_two_pow (n : ℕ) :
  ∃ p, nat.prime p ∧ odd p ∧ p ∣ n ∨ (∃ k : ℕ, n = 2 ^ k) :=
```
It also makes sense to keep the `2` stuff before the odd primes, because `2 < p` for odd `p`. But then the name should change (and the PR title).